apkid/rules/elf/protectors.yara

```diff
@@ -162,3 +162,45 @@ rule vkey_elf : protector
   condition:
     is_elf and $libname and 1 of ($vos*) and 1 of ($detection*) and 1 of ($jni*)
 }
+
+rule verimatrix_elf_arm64 : protector
+{
+  meta:
+    description = "InsideSecure Verymatrix"
+    url         = "https://www.verimatrix.com/products/app-shield/"
+    sample      = "105a2646e8acf45afcd606f4d47af68a8218e0ae7d8c9646d0048cbf32df7a73"
+    author      = "FrenchYeti"
+    
+  strings:
+    // byte sequence from .data, used into JNI_OnLoad
+    $seq = {
+      ?? ?? ?? ?? 88 98 a8 b8
+      ?? ?? ?? ?? 94 a4 b4 c4
+      ?? ?? ?? ?? 88 98 a8 b8
+      ?? ?? ?? ?? 94 a4 b4 c4
+    }
+    
+    $comm1 =  "Android (7019983 based on r365631c3) clang version 9.0.9 (https://android.googlesource.com/toolchain/llvm-project a2a1e703c0edb03ba29944e529ccbf457742737b) (based on LLVM 9.0.9svn)"
+    $comm2 = "GCC: (GNU) 4.9.x 20150123 (prerelease)"
```

Comment:
same here with regexes

Reply:
Not sure if this metavariable is very unique to the obfuscator. This tends to change over time.

Reply:
What do you suggest ? Be aware this string is the same into all tested samples

Reply:
I'd suggest to create 2 rules; one to catch this specific version and other adding InsideSecure Verimatrix (unknown version)

Reply:
Pattern removed. Rule will be updated later if required, after more tests if some false positive occur

Reply:
Hold your horses! Why dont you leave this and write another rule like this?

```sh
rule verimatrix_arm64_unknown : protector
{
  meta:
    description = "InsideSecure Verymatrix (potentially)"
    url         = "https://www.verimatrix.com/products/app-shield/"
    sample      = "???????"
    author      = "FrenchYeti"
    
  strings:
    // Put here your "fixed" compiler metadata
    $compiler_md = "xxxxxxx"
  
  condition:
    elf.machine == elf.EM_AARCH64 
    and verimatrix_arm64 and all of them
}
```

Reply:
Too much results with a rule based on both strings compared to the rule of this PR (see [https://koodous.com/my_rulesets/8177](https://koodous.com/my_rulesets/8177) ). I think it is better to remove both patterns.